src/main/java/com/google/ehub/data/EntertainmentItemDatastore.java

```diff
@@ -0,0 +1,153 @@
+package com.google.ehub.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.ehub.utility.NextIncreasingLexicographicStringUtility;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Singleton class that manages EntertainmentItems stored in Datastore.
+ */
+public final class EntertainmentItemDatastore {
+  private static final String ENTERTAINMENT_ITEM_KIND = "entertainmentItem";
+  private static final String TITLE_PROPERTY_KEY = "title";
+  private static final String LOWERCASE_TITLE_PROPERTY_KEY = "lowercaseTitle";
```

Comment:
also btw, the more standard terms for what you're doing is would be:
DISPLAY_TITLE and NORMALIZED_TITLE

Reply:
I updated the usage of this terms in last commit.